quic/s2n-quic-events/src/output/metrics/aggregate.rs

```diff
@@ -201,6 +232,43 @@ pub fn emit(output: &Output, files: &[File]) -> TokenStream {
                     ));
                 }
             }
+
+            for measure in field.attrs.measure_counter.iter() {
+                let name = format!("{snake}.{}", measure.name.value());
+                let units = measure.unit.as_ref().unwrap_or(&units_none);
+                let info = &info.push(&name, units);
+                let id = measures.push(info, None);
+                let ctx_name = Ident::new(&format!("ctr_{id}"), Span::call_site());
+
+                let value = quote!(event.#field_name.as_u64());
+                let counter_increment = output.mode.counter_increment_by(value);
+
+                context_fields.extend(quote!(
+                    #ctx_name: #counter_type,
+                ));
+                context_init.extend(quote!(
+                    #ctx_name: #counter_init,
+                ));
+
+                on_event.extend(quote!(
+                    context.#ctx_name #counter_increment;
+                ));
+
+                if units == "Duration" {
+                    on_connection_closed.extend(quote!(
+                        self.measure(#info, #id, core::time::Duration::from_micros(context.#ctx_name #counter_load));
```

Comment:
Hm, I guess I missed before that we have u64 granularity of microseconds in our metrics. Maybe we should switch to nanoseconds? Anyway, not important for this PR.

Reply:
Yeah we do currently:

https://github.com/aws/s2n-quic/blob/0dfeff7a5b2e92562daae01960874a156ca64886/quic/s2n-quic-core/src/event/metrics/aggregate/metric.rs#L120

The tricky bit is our timestamps aren't more granular than that either, so it would be a bit of a change: https://github.com/aws/s2n-quic/blob/0dfeff7a5b2e92562daae01960874a156ca64886/quic/s2n-quic-core/src/time/timestamp.rs#L112